### DISCUSSION TITLE
[bug?] useContractReads with enabled:false returning data

### DISCUSSION THREAD

**Discussion by danilo-89** [wevm/wagmi#635] [Q&A]
Is this a bug?

This code will console.log data from contract regardless of 'enabled:false' property?

```
export function ContractReads() {
	const { data, isError, isLoading } = useContractReads({
		contracts: [
			{
				addressOrName: '0xecb504d39723b0be0e3a9aa33d646642d1051ee1',
				contractInterface: abi,
				functionName: 'getAlive',
			},
			{
				addressOrName: '0xecb504d39723b0be0e3a9aa33d646642d1051ee1',
				contractInterface: abi,
				functionName: 'getBoredom',
			},
		],
		enabled: false,
	});

	console.log(data);

	return <div></div>;
}
```

**Comment by jeromevvb** [comment]
Same issue
Even with enabled set to false, the hook will perform the request...

**Comment by tmm** [comment]
Setting `enabled: false` works for me. It's possible the value is cached so `data` is defined and that is what you are seeing while logging.

**Reply by meri-maki** [reply]
Hi! If `enabled: false`, how do you run contractRead when needed?

**Reply by meri-maki** [reply]
@tmm hi! if ```enabled: false``` I run ```refetch()``` and it doesn't work
Is it possible to run contractreads on demand?
